iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -261,6 +268,13 @@ public boolean nextKeyValue() throws IOException {
       while (true) {
         if (currentIterator.hasNext()) {
           current = currentIterator.next();
+          Configuration conf = context.getConfiguration();
+          if (HiveIcebergStorageHandler.isDelete(conf, conf.get(Catalogs.NAME))) {
+            if (current instanceof GenericRecord) {
+              PositionDeleteInfo pdi = IcebergAcidUtil.parsePositionDeleteInfoFromRecord((GenericRecord) current);
```

Comment:
Do we need a `GenericRecord` here?

Reply:
Can we reuse the pdi object to save some time for GC?

Reply:
> Do we need a GenericRecord here?
We need to have a GenericRecord here because it's straightforward to grab the positional delete info for each record from the GenericRecord. I haven't looked into how to grab it from a VectorizedRowBatch (or its parquet equivalent), but seemed more complicated at first glance. As for whether we need the instanceof check, that's a different question, I guess we don't, since we already assert that vectorization is off during compilation, so I can remove that extra check

Reply:
> Can we reuse the pdi object to save some time for GC?

Yes, I think it's a good idea to reuse it. We can make it mutable and the AcidUtil can just update its fields